jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -249,6 +251,20 @@ protected SpanBuilder(String operationName) {
       this.operationName = operationName;
     }
 
+    private void verifyStartTimeInMicroseconds() {
+      int digits = 0;
+      long subject = 1;
+      while (subject <= startTimeMicroseconds) {
+        digits++;
+        subject *= 10;
+      }
+      if (digits >= DIGITS_IN_MICROSECOND_TIMESTAMP) {
+        throw new IllegalArgumentException(String
+            .format("'startTimeMicroseconds' value %d contains only %d digits, while a minimum of %d is expected",
+                startTimeMicroseconds, digits, DIGITS_IN_MICROSECOND_TIMESTAMP));
+      }
+    }
+
```

Comment:
is this necessary?
if yes, shell we just check`startTimeMicroseconds ` with `1e16`, instead of a loop check?

Reply:
The test is necessary because otherwise you get weird unsupported behavior and I think it's a good idea to resolve it.
As for your suggestion - you're right - I'll fix it.

Reply:
Fixed